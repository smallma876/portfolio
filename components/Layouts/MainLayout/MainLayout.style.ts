import styled from "styled-components";

const Wrapper = styled.div``;

const Top = styled.div`
    height: 8vh;
`;

const Bottom = styled.div`
    height: 92vh;
`;

export const MainLayoutStyles = {
  Wrapper,
  Top,
  Bottom,
};
