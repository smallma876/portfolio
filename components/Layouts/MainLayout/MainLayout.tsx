import React, { FC, ReactNode } from "react";
import { MainLayoutStyles } from "./MainLayout.style";

interface MainLayoutProps {
  children: ReactNode;
  top: ReactNode;
}

const MainLayout: FC<MainLayoutProps> = ({
  children,
  top,
}: MainLayoutProps) => {
  return (
    <MainLayoutStyles.Wrapper>
      <MainLayoutStyles.Top>{top}</MainLayoutStyles.Top>
      <MainLayoutStyles.Bottom>{children}</MainLayoutStyles.Bottom>
    </MainLayoutStyles.Wrapper>
  );
};

export default MainLayout;
