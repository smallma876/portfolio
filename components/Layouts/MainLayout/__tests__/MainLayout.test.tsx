import { render, screen } from "@testing-library/react";
import MainLayout from "../MainLayout";

const renderComponent = async () => {
  const { container } = await render(<MainLayout top="top">Test</MainLayout>);

  return container;
};

describe("MainLayout", () => {
  it("should render match snapshot", () => {
    const container = renderComponent();
    expect(screen.getByText("top")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
