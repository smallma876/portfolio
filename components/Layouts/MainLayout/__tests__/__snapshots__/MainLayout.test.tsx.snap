// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout should render match snapshot 1`] = `Promise {}`;
